scipy/optimize/optimize.py

```diff
@@ -52,26 +52,29 @@
 
 
 class MemoizeJac(object):
-    """ Decorator that caches the value gradient of function each time it
+    """ Decorator that caches the value and gradient of function each time it
     is called. """
 
     def __init__(self, fun):
         self.fun = fun
         self.jac = None
+        self.value = None
```

Comment:
Could you make this a private variable, `self._value = None`?

Reply:
done. But `self.jac` should stay public?

Reply:
Yeah, just leave it as it is, even those `MemoizeJac` isn't public I don't like making backwards incompatible changes.

Reply:
ah, very good point.